subprojects/kotlin-dsl/build.gradle.kts

```diff
@@ -74,7 +74,10 @@ dependencies {
     testImplementation(project(":platform-native")) {
         because("BuildType from platform-native is used in ProjectAccessorsClassPathTest")
     }
-    testImplementation(project(":plugins"))
+    testImplementation(project(":plugins")) {
+        because("For JavaPluginConvention and ApplicationPluginConvention")
+    }
+    testImplementation(project(":platform-jvm"))
```

Comment:
It looks like we should be able to cut the test dependency on plugins by changing the tests to look for a custom convention or something.

Reply:
Used a custom convention to avoid needing to use the conventions from `:plugins`